### DISCUSSION TITLE
Uncaught unwind exception

### DISCUSSION THREAD

**Discussion by aharondavid** [emscripten-core/emscripten#16918] [Q&A]
What can be the cause of 'Uncaught unwind' exception below , there are no debugger call stack and I don't have any clue what is rasing that error :
![image](https://user-images.githubusercontent.com/41375082/167302940-3469f4bb-3632-4606-ac2a-67da04b8ffc2.png)

**Comment by sbc100** [comment]
The `unwind` exception is used to unwind the stack if you call `emscripten_set_main_loop` or `emscripten_unwind_to_js_event_loop` or `emscripten_exit_with_live_runtime` and also if you call `exit`.

This location in the `invoke_iiii` function is just re-throwing that excption.   If you set a breakpoint there perhaps you can see the original call stack?

Can you share all the settings that you are using?   The best way to do this is if you can share your full linker command line.

**Comment by aharondavid** [comment]
eventually, the SAFE_HEAP=1 does the work.
the crash is because exception on unaligned pointer access.
I am wandering  if the 'unaligned pointer access' is just because setting of SAFE_HEAP

![image](https://user-images.githubusercontent.com/41375082/167889027-2cec4e92-1faf-4a40-90f0-706baff014d0.png)

**Comment by sbc100** [comment]
Looks like you want SAFE_HEAP=2:

https://github.com/emscripten-core/emscripten/blob/4cf36b6df367326e4d54ab6004466efe7ed3f265/src/settings.js#L305-L314

**Comment by aharondavid** [comment]
Thank!
I wouldn't get over on it without that help.
